agent/pkg/oas/oas_generator.go

```diff
@@ -98,8 +101,14 @@ func (g *defaultOasGenerator) runGenerator() {
 	dataChan := make(chan []byte)
 	metaChan := make(chan []byte)
 
-	logger.Log.Infof("Querying DB for OAS generator with query '%s'", g.entriesQuery)
-	g.dbConn.Query(g.entriesQuery, dataChan, metaChan)
+	func() {
```

Comment:
why is this part inside a func?

Reply:
It's from @undera's commits. `Query` call and the retrieval from `dataChan`/`metaChan` should run in parallel like in [this example](https://github.com/up9inc/basenine#query-1). But I'm not sure why there is a closure instead of a Goroutine.

Reply:
So what can we do? I don't think it's a good idea to leave it like that

Reply:
I'm not familiar with this OAS generator feature. But I understand that `runGenerator` is a Goroutine. So, it's written in a slightly different way than [the original example](https://github.com/up9inc/basenine#query-1). Therefore removed the closure with https://github.com/up9inc/mizu/commit/d93e00076d9be042ea1fc42597bc0917a5f874fb